pkgs/desktops/gnome-3/misc/gnome-flashback/default.nix

```diff
@@ -80,6 +89,58 @@ in stdenv.mkDerivation rec {
       packageName = pname;
       attrPath = "gnome3.${pname}";
     };
+
+    mkSessionForWm = { wmName, wmLabel, wmCommand }: stdenv.mkDerivation {
+      name = "gnome-flashback-${wmName}";
+
+      buildCommand = ''
+        mkdir -p $out/libexec
+        cat << EOF > $out/libexec/gnome-flashback-${wmName}
+        #!${bash}/bin/sh
+
+        if [ -z \$XDG_CURRENT_DESKTOP ]; then
```

Comment:
Using `lib.writeScript` would allow us to avoid the need to escape this.

Reply:
How would be put `$out` into it with `writeScript`?

Reply:
It produces output which can be copied or linked.

Reply:
I know but we want to add `$out/share` into `XDG_DATA_DIRS`, but we can't use `$out` in a `writeScript`.

Reply:
Again, I understand that. The question is how you would do the following in the `writeScript` given that `$out` is not available:

`export XDG_DATA_DIRS=$out/share:...`

Reply:
Hmm, sorry, you are right. I was thinking about passing a placeholder but those do not really work with `writeScript`, I think.